docs/source/building-supergraphs/router.mdx

```diff
@@ -345,53 +365,56 @@ class DataSourceWithServerId extends RemoteGraphQLDataSource {
   // highlight-end
 }
 
-const gateway = new ApolloGateway({
-  supergraphSdl,
-  // highlight-start
-  buildService({ url }) {
-    return new DataSourceWithServerId({ url });
-  }
-  // highlight-end
-});
-
-const server = new ApolloServer({
-  gateway,
-  context() {
-    return { serverIds: [] };
-  },
-  plugins: [
+async function startApolloGateway() {
+  const gateway = new ApolloGateway({
+    supergraphSdl,
     // highlight-start
-    {
-      requestDidStart() {
-        return {
-          willSendResponse({ context, response }) {
-            // Append our final result to the outgoing response headers
-            response.http.headers.set(
-              'Server-Id',
-              context.serverIds.join(',')
-            );
-          }
-        };
-      }
+    buildService({ url }) {
+      return new DataSourceWithServerId({ url });
     }
     // highlight-end
-  ]
-});
+  });
+
+  const server = new ApolloServer({
+    gateway,
+    plugins: [
+      // highlight-start
+      {
+        requestDidStart() {
```

Comment:
Not new in this PR, but since AS3 both this and willSendResponse below should be declare as `async requestDidStart()`.  (It probably will technically work in JS without it but not in TS.)